### PR TITLE
[308] Update stream fastlane actions

### PR DESCRIPTION
**Pull request by alteral**
### 🔗 Issue Links

Resolves https://github.com/GetStream/ios-issues-tracking/issues/308

### 🎯 Goal

Try to solve a nasty, unpredictable fastlane crash in the testing pipeline (presumably, the simulator is stuck)

```bash
2023-02-02 15:23:58.171 xcodebuild[16883:44216] [MT] IDETestOperationsObserverDebug: 462.731 elapsed -- Testing started completed.
2023-02-02 15:23:58.171 xcodebuild[16883:44216] [MT] IDETestOperationsObserverDebug: 0.000 sec, +0.000 sec -- start
2023-02-02 15:23:58.171 xcodebuild[16883:44216] [MT] IDETestOperationsObserverDebug: 462.731 sec, +462.731 sec -- end
Testing failed:
	StreamChatUITestsAppUITests-Runner (19377) encountered an error (Early unexpected exit, operation never finished bootstrapping - no restart will be attempted. (Underlying Error: Test crashed with signal kill before starting test execution. If you believe this error represents a bug, please attach the result bundle at /Users/runner/work/stream-chat-swift/stream-chat-swift/fastlane/test_output/StreamChatUITestsApp.xcresult))

** TEST EXECUTE FAILED **
[15:24:07]: Exit status: 65
```

### 📝 Summary

- Reset the simulator before booting it on CI
- Boot simulator before fastlane will try doing it

### ☑️ Contributor Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] This change follows zero ⚠️ policy (required)
- [ ] This change should be manually QAed
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### 🎁 Meme

gif

### CODE REVIEW

File: fastlane/Allurefile [hunk 1]
```diff
@@ -17,7 +17,7 @@ end
 
 desc 'Create launch on Allure TestOps'
 lane :allure_launch do |options|
-  next unless is_check_required(sources: sources_matrix[:e2e])
+  # next unless is_check_required(sources: sources_matrix[:e2e])
```
Comment (nuno-vieira) [inline fastlane/Allurefile:20]: Do we want to comment this?

Comment (alteral) [reply fastlane/Allurefile:20]: Oh, good catch 🙈. I was testing around. Will revert this shortly

Comment (alteral) [reply fastlane/Allurefile]: Done

**Review by nuno-vieira — APPROVED** [review]
LGTM! ✅

**Pull request merged**